src/sentry/tsdb/snuba.py

```diff
@@ -1,13 +1,24 @@
 from __future__ import absolute_import
 
 import collections
+from copy import deepcopy
 import six
 
 from sentry.tsdb.base import BaseTSDB, TSDBModel
-from sentry.utils import snuba
+from sentry.utils import snuba, outcomes
 from sentry.utils.dates import to_datetime
 
 
+SnubaModelSettings = collections.namedtuple(
+    # `groupby` - the column in Snuba that we want to put in the group by statement
+    # `aggregate` - the column in Snuba that we want to run the aggregate function on
+    # `dataset` - the dataset in Snuba that we want to query
```

Comment:
nit:
I would put the dataset first. 
aggregate, groupby, and conditions are part of the snuba query, while the dataset is generally a separate concept (we can pass it as part of the URL)